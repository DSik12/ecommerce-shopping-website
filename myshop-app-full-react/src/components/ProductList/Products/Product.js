import React from 'react';
import { Link } from 'react-router-dom';
import UpdateQuantity from '../../shared/Cart/UpdateQuantity';

const Product = ({product}) => {
    
    return (
        <div className="product" id={`product${product.id}`}>
            <div className="image-box">
                <div className={`images${product.id}`}><img src={product.imageURLs[0].thumbnail} alt={product.imageURLs[0].altText} width="250" height="250" id={`thumbnail${product.id}`} />
                </div>
            </div>
            <div className="text-box">
                <h2 className="item"><Link to={`./${product.id}`} id={`productName${product.id}`}>{product.productName}</Link></h2>
                <h3 className="price" id={`productPrice${product.id}`}>Rs.{product.price}</h3>
                <UpdateQuantity product={product}/>
            </div>
        </div>
    )
}

export default Product;
