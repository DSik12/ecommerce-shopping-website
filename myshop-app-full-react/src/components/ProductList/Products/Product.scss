@import "../../shared/ScssBreakpoint/Breakpoint.scss";
@mixin browser-support() {
    -webkit-touch-callout: none;
    -webkit-user-select: none;
    -khtml-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}
@mixin align-center() {
    text-align: center;
    vertical-align: middle;
}

@mixin border-style() {
    border: 2px solid black;
}

.products {
    @include border-style();
    margin: 110px 10px 10px;
}
.products .product-heading {
    padding: 10px 0 0 20px;
}

.products-heading {
    padding-left: 3%;
}

.text-box button {
    background: linear-gradient(to bottom, #009ce5, #0096db);
    color: #fff;
    padding: 5px;
    width: 200px;
    border-radius: 3px;
}

.listing-section {
    text-align: center;
    align-items: center;
}

.item {
    a {
        text-decoration: none;
        font-size: 15px;
    }
}

.plus-button,
.minus-button {
    display: inline-block;
    border: 1px solid black;
    margin: 0;
    width: 20px;
    height: 26px;
    @include align-center();
    background: #eee;
    @include browser-support();
}

.plus-button,
.minus-button:hover {
    cursor: pointer;
}

.product {
    /* text-align: center; */
    border: 1px solid black;
    margin: 10px 20px;
    padding: 10px 0;
    transition: all 1s ease;
}

@media (min-width: $tablet-portrait) {
    .productList {
        display: flex;
        @include border-style();
        margin: 110px 5px 5px;
    }

    .products {
        width: 72%;
        border: none;
        margin: 10px;
    }

    .products-heading {
        display: none;
    }

    .product {
        width: 90%;
        height: 100%;
        display: inline-block;
        border-radius: 2%;
        margin: 5px auto;
        @include border-style();
    }

    .product {
        .image-box {
            height: 250px;
            overflow: hidden;
            border-radius: 2% 2% 0 0;
        }
        .images {
            border-radius: 2% 2% 0 0;
            transition: all 1s ease;
            display: flex;
            align-items: center;
            justify-content: center;
        }
        .text-box {
            width: 100%;
            border-radius: 0 0 2% 2%;
        }
        .item {
            clear: left;
            width: 100%;
            text-align: center;
        }
        .text-box button {
            border-radius: 4px;
            background-image: linear-gradient(to bottom, #009ce5, #0096db);
            font-size: 15px;
            width: 80%;
            margin: auto;
            justify-content: center;
            color: #fff;
        }
    }
    .text-box {
        display: flex;
        flex-direction: column;
        button {
            height: 30px;
        }
        .price {
            height: 30px;
        }
        .item {
            height: 100px;
            text-overflow: ellipsis;
            overflow: scroll;
        }
    }
    .plus-button,
    .minus-button {
        display: inline-block;
        border: 1px solid black;
        margin: 0;
        width: 20px;
        height: 26px;
        text-align: center;
        vertical-align: middle;
        background: #eee;
        @include browser-support();
    }

    .plus-button,
    .minus-button:hover {
        cursor: pointer;
    }
    .listing-section {
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-column-gap: 0; // TWO TIMES !
        grid-column-gap: 20px;
        grid-row-gap: 20px;
    }
}

@media (min-width: $laptop) {
    .plus-button,
    .minus-button {
        display: inline-block;
        border: 1px solid black;
        margin: 0;
        width: 20px;
        height: 26px;
        text-align: center;
        vertical-align: middle;
        background: #eee;
        @include browser-support();
    }

    .plus-button,
    .minus-button:hover {
        cursor: pointer;
    }
    .listing-section {
        display: grid;
        grid-template-columns: 1fr 1fr 1fr;
        grid-column-gap: 20px;
        grid-row-gap: 20px;
    }

    .product:hover {
        transform: scale(1.05);
        overflow: hidden;
        border-radius: 1%;
    }
}
