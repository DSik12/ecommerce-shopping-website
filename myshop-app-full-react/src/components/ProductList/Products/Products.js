import fetchUrls from '../../../config/config.json';
import Product from './Product';
import useFetch from '../../../config/useFetch';
import queries from '../productListConfig/productList.json';

const Products = ({ urlParam  }) => {
  if(!urlParam){
    urlParam = "/?"+queries.start0to9;
  }
  const products = useFetch(fetchUrls.productApi + urlParam, "GET");

  let productList = null;
  if (products.apiData !== null) {
    productList = products.apiData.map((product) => {
      return (<Product product={product} key={product.id}></Product>)
    });
  }
  
  return (
    <div className="listing-section" id="listingSection">

      {productList && productList.length > 0 ?
        productList
        :
        <div>
          No Products Found
        </div>
      }
    </div>
  )
}

export default Products;