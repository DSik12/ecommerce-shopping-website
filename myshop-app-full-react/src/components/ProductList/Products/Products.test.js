import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Products from './Products.js';

describe("Products Component", () => {
    it('renders the component properly', () => {
        const divEl = document.createElement('div');
        ReactDom.render(<BrowserRouter><Products/></BrowserRouter>, divEl);
    });

    it('should display products correctly', () => {
        document.body.innerHTML = `
        <div className="product" id="product1">
            <div className="image-box">
                <div className="image1"><img src="image1.png" alt="product1" width="250" height="250" id="thumbnail1" />
                </div>
            </div>
            <div className="text-box">
                <h2 className="item"><Link to="./product1" id="productName1">Apple-iPhone-13-pro-smartphone</Link></h2>
                <h3 className="price" id="productPrice1">Rs.84900</h3>
                <UpdateQuantity product="product"/>
            </div>
        </div>
        <div className="product" id="product2">
            <div className="image-box">
                <div className="image2"><img src="image2.png" alt="product2" width="250" height="250" id="thumbnail2" />
                </div>
            </div>
            <div className="text-box">
                <h2 className="item"><Link to="./product2" id="productName2">Samsung Galaxys S21 FE 5G</Link></h2>
                <h3 className="price" id="productPrice2">Rs.54600</h3>
                <UpdateQuantity product="product"/>
            </div>
        </div>
        `;
        require('./Products.js');
    });

    it('should have right Products component snapshot', () => {
        const tree = renderer.create(<BrowserRouter><Products/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});