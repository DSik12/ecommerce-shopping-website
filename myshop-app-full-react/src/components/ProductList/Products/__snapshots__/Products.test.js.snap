// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Component should have right Products component snapshot 1`] = `
<div
  className="listing-section"
  id="listingSection"
>
  <div>
    No Products Found
  </div>
</div>
`;
